components/ActionBar/index.js

```diff
@@ -1,281 +1,422 @@
-      display: 'none'
-    }
-  })
-}
-
-class ActionBar extends Component {
-  constructor(props) {
-    super(props)
+const ActionBar = ({
+  mode,
+  document,
+  t,
+  inNativeApp,
+  share,
+  download,
+  isCentered
+}) => {
+  const [pdfOverlayVisible, setPdfOverlayVisible] = useState(false)
+  const [fontSizeOverlayVisible, setFontSizeOverlayVisible] = useState(false)
+  const [shareOverlayVisible, setShareOverlayVisible] = useState(false)
+  const [podcastOverlayVisible, setPodcastOverlayVisible] = useState(false)
 
-    this.state = {
-      showShareOverlay: false,
-      showFontSizeOverlay: false
-    }
+  if (!document) {
+    return (
+      <div {...styles.topRow} {...(isCentered && { ...styles.centered })}>
+        {download && (
+          <IconButton
+            href={download}
+            Icon={MdFileDownload}
+            label={share.label || ''}
+            target='_blank'
+          />
+        )}
 
-    this.toggleShare = () => {
-      this.setState({
-        showShareOverlay: !this.state.showShareOverlay
-      })
-    }
+        <IconButton
+          label={share.label || ''}
+          Icon={ShareIOSIcon}
+          href={share.url}
+          onClick={e => {
+            e.preventDefault()
+            trackEvent(['ActionBar', 'share', share.url])
+            if (inNativeApp) {
+              postMessage({
+                type: 'share',
+                payload: {
+                  title: share.title,
+                  url: share.url,
+                  subject: share.emailSubject || '',
+                  dialogTitle: t('article/share/title')
+                }
+              })
+              e.target.blur()
+            } else {
+              setShareOverlayVisible(!shareOverlayVisible)
+            }
+          }}
+        />
+        {shareOverlayVisible && (
+          <ShareOverlay
+            onClose={() => setShareOverlayVisible(false)}
+            url={share.url}
+            title={share.overlayTitle || t('article/actionbar/share')}
+            tweet={share.tweet || ''}
+            emailSubject={share.emailSubject || ''}
+            emailBody={share.emailBody || ''}
+            emailAttachUrl={share.emailAttachUrl}
+          />
+        )}
+      </div>
+    )
+  }
 
-    this.toggleFontSize = () => {
-      this.setState({
-        showFontSizeOverlay: !this.state.showFontSizeOverlay
-      })
-    }
+  const meta = document && {
+    ...document.meta,
+    url: `${PUBLIC_BASE_URL}${document.meta.path}`
   }
+  const podcast =
+    (meta && meta.podcast) ||
+    (meta && meta.audioSource && meta.format && meta.format.meta.podcast)
+  const hasPdf = meta && meta.template === 'article'
+  const isDiscussion = meta && meta.template === 'discussion'
+  const emailSubject = t('article/share/emailSubject', {
+    title: document.title
+  })
+  const {
+    discussionId,
+    discussionPath,
+    discussionQuery,
+    discussionCount,
+    isDiscussionPage
+  } = getDiscussionIconLinkProps(
```

Comment:
Any idea why this function always returns "/dialog" instead of the link to the discussion? I didn't change the function, just copied it, like it was done in the previous component. @tpreusse